standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -2419,9 +2419,9 @@ private void create_table_core(final RawStore ms, final CreateTableRequest req)
       }
 
       if (!TableType.VIRTUAL_VIEW.toString().equals(tbl.getTableType())) {
```

Comment:
this conditional doesn't seem to only match "managed" tables - could that cause any issues?

with the "translator" in place I think we are mangling the location at too many places...

Reply:
only managed tables would have visibilityId poppulated